server/src/main/java/org/hl7/davinci/endpoint/rems/controller/RemsController.java

```diff
@@ -41,4 +72,87 @@ private ResponseEntity<Drug> processRequirements(Drug drug) {
                 .contentType(MediaType.parseMediaType(MediaType.APPLICATION_JSON_VALUE))
                 .body(drug);
     }
+
+    public void updateRemsRequestStatus(String uid) {
+        try {
+          TimeUnit.SECONDS.sleep(30);
+        }
+        catch(Exception e)
+        {
+            System.out.println(e);
+          }
+        Rems rems = remsRepository.findById(uid).orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, uid + " not found"));
+        rems.setStatus("Approved");
+        remsRepository.save(rems);
+      }
+    
+      public void updateRemsRequestStatusInBackground (final String uid) {
+        Thread t = new Thread(() -> updateRemsRequestStatus(uid));
+        t.start();
+      }
+    
+      @PostMapping(value = "/rems")
+      @CrossOrigin
+      public ResponseEntity<Object> postRems(@RequestBody String jsonData) {
+        JsonNode remsObject = JacksonUtil.toJsonNode(jsonData);
+        String id = UUID.randomUUID().toString().replace("-", "");
+
+        JsonNode params = getResource(remsObject, remsObject.get("entry").get(0).get("resource").get("focus").get("parameters").get("reference").textValue());
+        
+        String prescriptionReference = "";
+        for (JsonNode param : params.get("parameter")) {
+            if (param.get("name").textValue().equals("prescription")) {
+                prescriptionReference = param.get("reference").textValue();
+            }
+        }
+
+        JsonNode presciption = getResource(remsObject, prescriptionReference);
+        String prescriptionSystem = presciption.get("medicationCodeableConcept").get("coding").get(0).get("system").textValue();
+        String prescriptionCode = presciption.get("medicationCodeableConcept").get("coding").get(0).get("code").textValue();
+        Drug drug = drugsRepository.findDrugByCode(prescriptionSystem, prescriptionCode).get(0);
+
+
+
+        Rems remsRequest = new Rems();
+        remsRequest.setCase_number(id);
+        remsRequest.setStatus("Pending");
+        remsRequest.setResource(remsObject);
+        remsRepository.save(remsRequest);
```

Comment:
need to re-save after the for loop to save the change to the rems object when the metReq is added.